### PR TITLE
research field in audience composition

### PR DESCRIPTION
**Pull request by Drumor**
add a research field in audience composition in place of a select element

### CODE REVIEW

**Comment by Drumor** [comment]
linked #710

File: inginious/frontend/templates/course_admin/audience_edit.html [hunk 1]
```diff
@@ -66,15 +66,18 @@ <h4 class="modal-title">{{ _("Add student") }}</h4>
                                         <label class="control-label" for="registered_students"> {{ _("Choose student :") }}</label>
                                     </div>
                                     <div class="col-md-9">
-                                        <select id="registered_students" name="registered_students" class="form-control">
+                                        <div class="col-sm-10">
+                                            {{ user_selection_box(course_id=course.get_id(), current_users=other_students, name="registered_students", id="registered_students", placeholder=_("Enter something here to search for a user")) | safe }}
+                                        </div>
+                                        <!--<select id="registered_students" name="registered_students" class="form-control">
```
Comment (anthonygego) [inline inginious/frontend/templates/course_admin/audience_edit.html:72]: Please remove commented code here.

**Comment by anthonygego** [comment]
You will have to update the minified js too.

**Comment by Drumor** [comment]
Done

**Comment by anthonygego** [comment]
Can you solve the conflicts (regenerating the minified js) ?

**Comment by Drumor** [comment]
This one is ok to review/merge

**Review by anthonygego — CHANGES_REQUESTED** [review]
There is an improvement. However:
- the "registered students" tab actually lists all the available users.
- it is therefore possible to add an new student to register this way, but it won't be stored once the save button pushed.

=> Ideally, one box is enough. The unregistered student can be registered directly. Maybe we can warn the admin that the student is not part of the course students yet. This should simplify the js code as we are sure that users added in the interface are real INGInious users.

**Comment by Drumor** [comment]
> There is an improvement. However:
> 
> * the "registered students" tab actually lists all the available users.
> * it is therefore possible to add an new student to register this way, but it won't be stored once the save button pushed.
> 
> => Ideally, one box is enough. The unregistered student can be registered directly. Maybe we can warn the admin that the student is not part of the course students yet. This should simplify the js code as we are sure that users added in the interface are real INGInious users.

In fact part of the logic was already done, so i simplified frontend to have only one box. Right now the admin is not aware that a student is registered or not to the course. If not, he is silently add to the course.

**Comment by anthonygego** [comment]
> it is therefore possible to add an new student to register this way, but it won't be stored once the save button pushed.

This still does not work. How to reproduce :
- Add a student that is not registered to the course yet.
- Click on save, nothing changes.

File: inginious/frontend/templates/course_admin/audience_edit.html [hunk 2]
```diff
@@ -47,44 +47,16 @@ <h4 class="modal-title">{{ _("Add student") }}</h4>
...
-                                    <i class="fa fa-user fa-fw"></i>&nbsp; {{ _("Registered student") }}</a>
-                                </li>
-                                <li role="presentation" class="nav-item">
-                                    <a href="#tab_new_student" class="nav-link" aria-controls="new_student" role="tab" data-toggle="tab">
-                                    <i class="fa fa-plus fa-fw"></i>&nbsp; {{ _("New student") }}</a>
-                                </li>
-                            </ul>
-                        </div>
                         <div class="tab-content">
                             <div role="tabpanel" class="card-body tab-pane active" id="tab_registered_student">
```
Comment (anthonygego) [inline inginious/frontend/templates/course_admin/audience_edit.html:51]: You can also remove the tabpanel as there is only one option left.

**Comment by Drumor** [comment]
> > it is therefore possible to add an new student to register this way, but it won't be stored once the save button pushed.
> 
> This still does not work. How to reproduce :
> 
> * Add a student that is not registered to the course yet.
> * Click on save, nothing changes.

Seems ok to me no?

https://user-images.githubusercontent.com/13468630/146527043-018549ab-2d6d-486f-ad22-ab7184566fff.mp4

**Review by anthonygego — APPROVED** [review]

**Comment by anthonygego** [comment]
LGTM

**Pull request merged**